pgrollup/parser_pg.py

```diff
@@ -0,0 +1,290 @@
+import sys
+from lark import Lark, Transformer
+
+def parse_create(sql):
+    '''
+    Example expected output:
+    [{'joininfos': '[{"table_name": "testjoin1", "table_alias": "t1", "condition": "", "join_type": "FROM"}, 
+    {"table_name": "testjoin2", "table_alias": "t2", "condition": "using (id)", "join_type": "INNER JOIN"}]',
+    'rollup_name': 'testjoin_rollup1',
+    'groups': [['t1.name', '"t1.name"']],
+    'columns': [['sum(t1.num)', 'sum_num'], ['sum(t2.foo)', 'sum_foo']],
+    'where_clause': None, 'having_clause': None}]
+    '''
+    dict = parse(sql)
+    infos = []                                              
+    infos.append({
+        'joininfos' :_getjoins(dict[0]['stmt']['fromClause'][0]),
+#       'groups' : [[x['fields'][0],str(x['fields'][0])] for x in dict[0]['stmt']['groupClause']],
+        'groups' : _getgroups(dict),
+        'columns': [[expr_to_str(x['val']),_getalias(x)] for x in dict[0]['stmt']['targetList']],
+        'where_clause': _getwhere(dict),
```

Comment:
as far as I can tell, this is the only place that the `_getwhere` function is used; this function is too simple to be separated out into it's own function, and instead should be included inline here; the function call can be replaced with the following code:
```
sql[0]['stmt'].get('whereClause','')
```

Also, the fact that you're hardcoding the index `[0]` seems like its very likely to have bugs if there's more than 1 sql statement


----

this same comment applies to (at least) the `_getalias` and `_gethaving` functions as well